src/main/java/com/carmen/app/control/PersistenceService.java

```diff
@@ -1,54 +1,79 @@
 package com.carmen.app.control;
 
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import javax.ejb.Stateless;
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
 import javax.persistence.TypedQuery;
-
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Expression;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
 
 import com.carmen.app.utils.Logged;
+
 /**
  * 
- *Service that generate a CRUD to be used by any resource
+ * Service that generate a CRUD to be used by any resource
+ * 
  * @author Carmen Pi�era Moreno
  *
  * @param <T> refers to a type
  * @param <K> refers to a key
  */
+
 @Stateless
 @Logged
-public class PersistenceService<T,K>{
+public class PersistenceService<T, K> {
 
-	
 	@PersistenceContext(unitName = "carP")
 	private EntityManager em;
-	
-	public List<T> getAll(String NameQuery, Class<T> c){
-		TypedQuery<T> query = this.em.createNamedQuery(NameQuery, c);
-		return query.getResultList();
+
+	public TypedQuery<T> getAll(Class<T> c, Map<String, String> filterMap, String orderBy) {
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<T> cq = cb.createQuery(c);
+		Root<T> r = cq.from(c);
+		cq.select(r);
+		List<Predicate> predicates = new ArrayList<Predicate>();
```

Comment:
Usually namedQueries are more clear in terms of code readability. The are much closer to SQL.